### DISCUSSION TITLE
Unifi OS 3.0.20 - umm-iptv

### DISCUSSION THREAD

**Discussion by jZNJ** [fabianishere/udm-iptv#256] [Q&A]
Hi,

![Screen Shot 2023-05-04 at 07 44 53](https://user-images.githubusercontent.com/40464359/236121763-ec2a5b7a-bc04-4cf2-96ba-d55f1bdf4c46.png)

I have KPN TV working perfectly thanks to the script that was provided here.

My question.
This will be the first time I have to do the Unifi OS upgrade when the udm-iptv script is installed.
Will the script for udm-iptv work for this Unifi OS version?

What prerequisitons should I do before upgrading the OS?

From what I read and understand is:
- Make a backup of 'the configuration'.  But I have  made no configurations. So expect no backup is needed.
- Do the Unifi OS upgrade.
- Check if KPN TV is working. I expect not because see several notes that the script need to be applied again.
- If not working run the script again.

Am I right to do it this way?

Thanks in advance,
John

**Comment by jZNJ** [comment]
note: or should I also do before the Unifi OS upgrade: udm-iptv uninstall

**Comment by julianbl** [comment]
Hi! I've just updated to 3.0.20. I was not able to make iptv work natively but reinstalling udm-iptv as always, works perfectly!

**Reply by jZNJ** [reply]
Thanks for your reply!
About reinstall: uninstall (remove) and do the install?

**Reply by julianbl** [reply]
No need, the OS upgrade removes everything. So when you are install, it's like a new one.

**Reply by jZNJ** [reply]
Thank you!

**Reply by Edu-ST** [reply]
> Hi! I've just updated to 3.0.20. I was not able to make iptv work natively but reinstalling udm-iptv as always, works perfectly!

Hi @julianbl    I tried reinstall udm-iptv after update to 2.5.17, but I have got this message
unifi-os: command not found , on 2.5.17

how did you do to install directly on this your 3.0.20 version?  could you support me please !!!
thank you !!!

**Reply by jZNJ** [reply]
These are my notes for the update.
Note: after the update of Unifi OS and executing the udm-iptv script: I had to reboot the KPN receiver. Because there was a message that the box could not connect to the modem.


Before doing the Unifi OS update:
- Checked if I was able to ssh to the router
- Checked in application Network which network I use for IPTV:
![Screen Shot 2023-05-08 at 15 46 00](https://user-images.githubusercontent.com/40464359/236840653-e7d461ee-759a-4f97-af9e-cdb47e4f9a9b.png)

Doing the Unifi OS update
![Screen Shot 2023-05-08 at 15 50 20](https://user-images.githubusercontent.com/40464359/236841678-24c1b53c-857e-486c-ac11-7c34f58ac0f4.png)

![Screen Shot 2023-05-08 at 15 50 36](https://user-images.githubusercontent.com/40464359/236841794-7e43e556-8619-4492-a930-86293c08a323.png)

![Screen Shot 2023-05-08 at 15 51 09](https://user-images.githubusercontent.com/40464359/236841892-1a8daafa-524f-432c-b5f2-7eed6fb50453.png)


When finished: SSH to the router
![Screen Shot 2023-05-08 at 15 47 24](https://user-images.githubusercontent.com/40464359/236840985-5872bc7b-aced-4f7c-b65e-74717573fdbd.png)


- Execute the script: sh -c "$(curl https://raw.githubusercontent.com/fabianishere/udm-iptv/master/install.sh -sSf)"

- Had the following dialogues
![Screen Shot 2023-05-08 at 15 53 27](https://user-images.githubusercontent.com/40464359/236842436-ef6a9643-96d0-4e33-bc0d-605a1abd918b.png)

**Comment by jZNJ** [comment]
@fabianishere 

Hi,

I updated my udm pro to  3.1.16 and completed with ssh
sh -c "$(curl https://raw.githubusercontent.com/fabianishere/udm-iptv/master/install.sh -sSf)"

The unifi os updates before went without issues, except this time.
After restarting the KPN receiver there is a message: stb-nmc-400

I have this from udm-iptv diagnose


Please share the following output with the developers:
=== Configuration ===
WAN Interface: eth8
WAN VLAN: 4 (dev iptv)
WAN DHCP: true (options "-O staticroutes -V IPTV_RG")
WAN Ranges: 213.75.0.0/16  217.166.0.0/16
LAN Interfaces: 
IGMP Proxy quickleave disabled: false
IGMP Proxy debug: false
=== IP Link and Route ===
40: iptv@eth8: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    inet 10.166.213.142/19 brd 10.166.223.255 scope global iptv
       valid_lft forever preferred_lft forever
10.166.192.0/19 proto kernel scope link src 10.166.213.142 
213.75.112.0/21 via 10.166.192.1 metric 240 
=== Service Logs ===
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: Obtaining IP address for VLAN interface
Sep 09 13:09:52 UDMPro udm-iptvd[2815]: udhcpc: started, v1.30.1
Sep 09 13:09:52 UDMPro udm-iptvd[2815]: udhcpc: sending discover
Sep 09 13:09:52 UDMPro udm-iptvd[2815]: udhcpc: sending select for 10.166.213.142
Sep 09 13:09:52 UDMPro udm-iptvd[2815]: udhcpc: lease of 10.166.213.142 obtained, lease time 3600
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: NATing IPTV network ranges (if necessary)
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: Creating static routes (if necessary)
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: Setting up IGMP Proxy
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: Using igmpproxy...
Sep 09 13:09:52 UDMPro udm-iptvd[2805]: Starting IGMP Proxy


Hope you can help me with this.

Thanks,
John

**Reply by jZNJ** [reply]
@fabianishere 

I see in udm-iptv.conf

that is missing:
IPTV_LAN_INTERFACES="br4"

in my idm-iptv.conf there is:
IPTV_LAN_INTERFACES=""

**Comment by jZNJ** [comment]
@fabianishere 
Sorry for the panic. I corrected the .conf by inserting br4 and now it works! (After switching the KPN receiver off and on).
